streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala

```diff
@@ -176,15 +176,17 @@ class KStream[K, V](val inner: KStreamJ[K, V]) {
     inner.foreach((k: K, v: V) => action(k, v))
 
   /**
-   * Creates an array of {@code KStream} from this stream by branching the records in the original stream based on
+   * Creates an array of `KStream` from this stream by branching the records in the original stream based on
    * the supplied predicates.
    *
    * @param predicates the ordered list of functions that return a Boolean
    * @return multiple distinct substreams of this [[KStream]]
    * @see `org.apache.kafka.streams.kstream.KStream#branch`
    */
-  def branch(predicates: (K, V) => Boolean*): Array[KStream[K, V]] =
+  //noinspection ScalaUnnecessaryParentheses
+  def branch(predicates: ((K, V) => Boolean)*): Array[KStream[K, V]] = {
```

Comment:
Why this change?

Reply:
The type checker in IDEA was spazzing out and treating `predicates`'s type as `Seq[(K,V) => Any]`.
This was the only way I could get it to shut up.

I actually couldn't compile this class in IDEA without this change.

Funny enough, it fails without the parens and then complains that the parens are unnecessary once you add them.

Can you check in your IDEA that you get no reported error and that you can run the tests I added from IDEA? If so, maybe I have the ide misconfigured.